pkg/sql/catalog/tabledesc/table_desc_builder.go

```diff
@@ -192,11 +192,13 @@ func maybeFillInDescriptor(
 
 	changes.UpgradedIndexFormatVersion = maybeUpgradePrimaryIndexFormatVersion(desc)
 	for i := range desc.Indexes {
-		changes.UpgradedIndexFormatVersion = changes.UpgradedIndexFormatVersion || maybeUpgradeSecondaryIndexFormatVersion(&desc.Indexes[i])
```

Comment:
Assuming a table descriptor has several secondary indexes needing an upgrade, short-circuit evaluation would ensure that only the first secondary index would get upgraded 😬 

I wrote this. I'm quite embarrassed.

Reply:
Oh man. Glad you caught this. Does this have any implications for needing to retain migrations we might have stripped?

Reply:
I don't think so and I wouldn't worry about it. If we need to worry about it, then let's think about putting in a "standing migration" of sorts which round-trips each descriptor in `system.descriptor` through `RunPostDeserializationChanges` at the end of each major version upgrade.